pkg/tools/clienturlctx/context.go

```diff
@@ -50,3 +51,22 @@ func ClientURL(ctx context.Context) *url.URL {
 	}
 	return nil
 }
+
+// WithDNSServerURLs -
+//    Wraps 'parent' in a new Context that has the DNS server URLs
+func WithDNSServerURLs(parent context.Context, urls []url.URL) context.Context {
+	if parent == nil {
+		panic("cannot create context from nil parent")
+	}
+	log.FromContext(parent).Debugf("passed DNS server urls: %v", urls)
+	return context.WithValue(parent, dnsServerURLSKey, urls)
+}
+
+// DNSServerURLs -
+//    Returns  the DNS server URLs
+func DNSServerURLs(ctx context.Context) []url.URL {
```

Comment:
```suggestion
func ClientURLs(ctx context.Context) []url.URL {
```

Reply:
@NikitaSkrynnik Also lets add a check if we dont have things by clienturl**s**Key but have something by clienturlurlKey then return it as a single value slice?